sqldelight-gradle-plugin/src/test/integration-mysql-async/src/test/kotlin/app/cash/sqldelight/mysql/integration/MySqlTest.kt

```diff
@@ -0,0 +1,90 @@
+package app.cash.sqldelight.mysql.integration
+
+import app.cash.sqldelight.driver.r2dbc.R2dbcDriver
+import com.google.common.truth.Truth.assertThat
+import io.r2dbc.spi.ConnectionFactories
+import kotlinx.coroutines.reactive.awaitSingle
+import org.junit.Test
+
+class MySqlTest {
+  private val factory = ConnectionFactories.get("r2dbc:tc:mysql:///myDb?TC_IMAGE_TAG=8.0")
+
+  private fun runTest(block: suspend (MyDatabase) -> Unit) = kotlinx.coroutines.test.runTest {
+    val db = before()
+    block(db)
+  }
+
+  suspend fun before(): MyDatabase {
+    val connection = factory.create().awaitSingle()
+    val driver = R2dbcDriver(connection)
+
+    return MyDatabase(driver).also { MyDatabase.Schema.create(driver) }
+  }
+
+  fun after() {
+  }
+
+  @Test fun simpleSelect() = runTest { database ->
+    database.dogQueries.insertDog("Tilda", "Pomeranian", true)
+    assertThat(database.dogQueries.selectDogs().executeAsOne())
+      .isEqualTo(
+        Dog(
+          name = "Tilda",
+          breed = "Pomeranian",
+          is_good = true
+        )
+      )
+  }
+
+  @Test
+  fun simpleSelectWithIn() = runTest { database ->
+    with(database) {
+      dogQueries.insertDog("Tilda", "Pomeranian", true)
+      dogQueries.insertDog("Tucker", "Portuguese Water Dog", true)
+      dogQueries.insertDog("Cujo", "Pomeranian", false)
+      dogQueries.insertDog("Buddy", "Pomeranian", true)
+      assertThat(
+        dogQueries.selectDogsByBreedAndNames(
+          breed = "Pomeranian",
+          name = listOf("Tilda", "Buddy")
+        ).executeAsList()
+      )
+        .containsExactly(
+          Dog(
+            name = "Tilda",
+            breed = "Pomeranian",
+            is_good = true
+          ),
+          Dog(
+            name = "Buddy",
+            breed = "Pomeranian",
+            is_good = true
+          )
+        )
+    }
+  }
+
+  /* TODO: Dates are currently broken with the mysql r2dbc driver
```

Comment:
whats the exception?

Reply:
and i'd prefer just deleting the code over adding commented code

Reply:
If I remember correctly the mysql driver can't return an `OffsetDateTime` which is expected. It looks like a future release of the driver will [return `ZonedDateTime` by default](https://github.com/mirromutth/r2dbc-mysql/pull/210) which I guess won't fix the issue, but it's also been almost two years since the last release.

I'll just remove this test and we can revisit it again if it comes up in the future.